### ISSUE TITLE
IoT Edge Storage Blob ARM64v8 - InvalidHeaderValue

### ISSUE THREAD

**Issue by flashgordonmjg** [Azure/azure-sdk-for-python#27925]
**_Device_** 
OS: Ubuntu 20.04.5 LTS
IoT Edge Runtime: Version - 1.2.10.55800403

**_Client Container_** Module:
- **Package Name**:  azure-storage-blob
- **Package Version**:  12.14.1
- **Operating System**:  **arm64v8**/python:3.7-slim-buster
- **Python Version**:  3.7.16
- **Package Name**:  azure-iot-device
- **Package Version**: 2.7.0

**_Blob Container_** Image:
- **Package Name**:  azure-storage-blob
- **Package Version**:  mcr.microsoft.com/azure-blob-storage:1.2-linux-arm64v8

**Describe the bug**
I have reviewed several issues regarding InvalidHeaderValue but I am still cannot successfully connect and create a blob container when calling azure-blob-storage:1.2-linux-arm64v8 container.  I am getting the error:
```
<?xml version="1.0" encoding="utf-8"?><Error><Code>InvalidHeaderValue</Code><Message>The value for one of the HTTP headers is not in the correct format. RequestId:ecc2b26b-2188-4102-9ed2-f9d7ab5c5e47 Time:2022-12-13T16:36:32.2146329Z</Message><HeaderName>x-ms-version</HeaderName><HeaderValue>2019-02-02</HeaderValue><ExceptionDetails><ExceptionMessage>The value 2019-02-02 provided for request header x-ms-version is invalid.</ExceptionMessage>
```

I have tried all of the displayed  dates from 'Please select from: ' in 1st Step below  by setting  each date in the `api_version` field:
```python
container_client  =  ContainerClient.from_connection_string(conn_str=blob_conn_str, container_name=container_name, logging_enable=enable_logs, api_version="2019-07-07")
```

**To Reproduce**

1. First, i set the date in the `api_version`="2019-01-01" and received back:
```
Exception: Unsupported API version '2019-01-01'. 
Please select from: 2019-02-02 2019-07-07 2019-10-10 2019-12-12 2020-02-10 2020-04-08 2020-06-12 2020-08-04 2020-10-02 2020-12-06 2021-02-12 2021-04-10 2021-06-08 2021-08-06
```
2. I then tried all of the dates in the `api_version` field and every the same message is returned with the exception of the date that was sent in the `api_version` field.
 
**Expected behavior**
Expectation is that a container is created when calling mcr.microsoft.com/azure-blob-storage:1.2-linux-arm64v8

**Additional context**
I did review [azure-blob-stroage container versions](https://mcr.microsoft.com/v2/azure-blob-storage/tags/list) and found that 1.2-linux-arm64v8 is tag the latest for arm64v8.  I suspect this is an issue of a new version of azure-storage-blob (in the client python _Client Container_) calling an older version of the azure-storage-blob (in the _Blob Container_)  api in the container but I am not sure how to fix.  I also have tried different versions of the azure-storage-blob in the Client Container with no success.

Thank you for the help.

**Comment by kristapratico** [comment]
Thanks for the detailed report, we'll take a look as soon as possible.

**Comment by jalauzon-msft** [comment]
Hi @flashgordonmjg Mark, I'm not too familiar with the IoT Edge Storage but this does seem like an issue with a newer Blob client version calling into a seemingly very old version of the Storage on IoT edge. It would seem from the `x-ms-version` header being invalid even for version `2019-02-02`, that the IoT Edge Storage service does not support any version that the modern versions of the Storage SDK support, as `2019-02-02` is the earliest our current (v12.x.x) SDKs support.

I unfortunately don't have a great recommendation here if you say that the `mcr.microsoft.com/azure-blob-storage:1.2-linux-arm64v8` container is the latest available. Looking at [this documentation](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-store-data-blob?view=iotedge-1.4#azure-blob-storage-quickstart-samples) and specifically the .NET recommendation, it looks like IoT Edge V1.4.1 should support the later service versions as it does say it will work with .NET SDK versions 12.8.0 (meaning it should work with at least earlier versions of the V12 Python SDK or any version if you specify the `api_version`). Is it possible to update your IoT Edge runtime version and then try and get a more up to date Storage container? If you're not able to find what you're looking for there, it's probably best to bring that up with the IoT Edge team via a Support request or GitHub issue if their software is OSS.

As a last resort, you may be able to try to downgrade your Python client version to [2.1.0](https://pypi.org/project/azure-storage-blob/2.1.0/) as that version will support even older service versions. However, **this is not recommended** as anything before V12 is considered deprecated. The library also went through quite a few changes to the structure with many breaking changes so code written against V12.X.X will not just work with 2.1.0. But again, I would try to find or get an updated version of IoT Edge Storage before going down this route.

Hopefully this helps, thanks.

**Comment by flashgordonmjg** [comment]
Hi @jalauzon-msft , I want to use Python so I can store images using OpenCV Python library into blob storage.  As far as the container is concerned, I am using Microsoft's supported containers from the list [Microsoft Container Registry](https://mcr.microsoft.com/v2/azure-blob-storage/tags/list). Reading the documentation, it appears that Azure Blob Storage container should run on ARM64.  I couldn't find any documentation that said ARM64 is not supported, rather it is supported.

That said, the indication that it might not is when reviewing [Microsoft Docker Hub - Azure Blob Storage](https://hub.docker.com/_/microsoft-azure-blob-storage?tab=description). From what I can tell in the Release Notes, the only version that supports ARM64 is Azure Blob Storage on IoT Edge 1.2.0 -Preview version .  Is there a reason why Microsoft doesn't support ARM64  in the newer versions of azure-blob-storage container?

I cannot imagine that Microsoft doesn't support Azure Blob Storage container running on ARM64 which makes me think that I am missing something very obvious.

What are my next steps to figure out how to get Microsoft supported Azure Blob Storage container running and working with azure-storage-blob client module?  Do I post this in under a different repo?

Thank you.

**Comment by jalauzon-msft** [comment]
Hi @flashgordonmjg Mark, looking at the release notes in [this](https://hub.docker.com/_/microsoft-azure-blob-storage?tab=description) document you linked, it does seem like the Blob Storage container has not been released for ARM64 since 1.2.0-Preview version. We do not work on this product so unfortunately; I don't have information to share on why that is the case.

You can try using the feedback options linked on the Docker page ([here](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-store-data-blob?view=iotedge-1.4#feedback)) to investigate further or to submit an issue to ask when support for ARM64 is coming.

Sorry I don't have a better answer for you on this. Thanks.

**Comment by flashgordonmjg** [comment]
Hi @jalauzon-msft Jacob, ok. I understand.
